lib/train/extras/command_wrapper.rb

```diff
@@ -162,7 +162,13 @@ class CommandWrapper
     include_options WindowsCommand
 
     def self.load(transport, options)
-      if transport.os.unix?
+      if transport.class == Train::Transports::SSH::Connection && !transport.connected?
+        # Need to call run_command here to get the message and show it to user while raising exception
+        msg = transport.run_command("whoami").stdout
+        # this will throw the error as
+        # Train::UserError: Ssh failed: Please login as the user "username" rather than the user "root".
+        raise Train::UserError.new("Ssh failed: #{msg}")
+      elsif transport.os.unix?
```

Comment:
So, my understanding of what this code is trying to do is to take the situation where the OS detection failed (because of auth/connection issues) but provide an error message that makes that more clear. I think it makes more sense from a design / logical flow perspective to push that down to a rescue on this method. IOW, react to the bad condition, rather than make it part of the normal checks:

```ruby
def self.load(transport, options)
  if transport.os.unix?
    # ...
rescue Train::PlatformDetectionFailed => e
  transport.valididate_connection # either raises or:
  raise e
end
```

This would require `valididate_connection` to be an abstract method (stubbed out in the superclass and raises if not overridden) on Transport::BaseConnection and on all supported Transports, which might be out of scope for this PR (not my call).